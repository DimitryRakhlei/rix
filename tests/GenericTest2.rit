import "rsl/Vector.rit"
#a = Vector{String}(100)

#i.for (0,10)
    a[i] = "i = " + i + " -> " + (i+2).sqrt()

i.for (4,9)
    print (a[i])
