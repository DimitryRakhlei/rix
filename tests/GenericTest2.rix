import ("rsl/Vector.rix")

loadFactor->String(Vector v) = (v.length() + "/" + v.capacity())

#a = Vector{String}(100)
#i.for (0,10)
    a[i] = "i = " + i + " -> " + (i+2).sqrt()

i.for (4,9)
    print (a[i])

print ("a is " + loadFactor(a))

if (a.isNull(9))
    print ("No element at index 9")
else
    print ("Element exists at index 9")


#b = Vector{int}(100)
i.for (0,8)
    b[i] = i

i.for (4,9)
    print (b[i])

#bLoadFactor = loadFactor(b)

print ("b is " + bLoadFactor)

if (b.isNull(9))
    print ("No element at index 9")
else
    print ("Element exists at index 9")