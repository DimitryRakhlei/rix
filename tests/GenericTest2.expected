i = 4 -> 2
i = 5 -> 2
i = 6 -> 2
i = 7 -> 3
i = 8 -> 3
a is at load 10/100
Element exists at index 9
4
5
6
7
0
b is 8/100
No element at index 9
